tests/unit/all_shortest_paths.c

```diff
@@ -46,42 +48,53 @@ int main() {
 
     from = 0; to = 0;
 
-    igraph_get_all_shortest_paths(&graph, &paths, &nrgeo, from, igraph_vss_1(to), IGRAPH_ALL);
+    igraph_get_all_shortest_paths(&graph, &paths, NULL, &nrgeo, from, igraph_vss_1(to), IGRAPH_ALL);
 
     for (i=0; i < igraph_vector_ptr_size(&paths); ++i) {
         print_vector(VECTOR(paths)[i]);
+        igraph_vector_destroy(VECTOR(paths)[i]);
+        igraph_free(VECTOR(paths)[i]);
+        VECTOR(paths)[i] = 0;
     }
     IGRAPH_ASSERT(igraph_vector_ptr_size(&paths) == VECTOR(nrgeo)[to]);
 
-    igraph_vector_ptr_free_all(&paths);
```

Comment:
I am wondering why this needed to be removed. The test _should_ work with `vector_ptr_free_all`.

Importantly, the `get_all_shortest_paths` function should not change the item destructor of `path` (if it does, that must be corrected).

Reply:
I did it because when I use `vector_ptr_free_all` I get memory leak. So I used the same destroy mechanism as in `igraph_get_all_shortest_paths_dijkstra.c`  tests.

Reply:
> I did it because when I use `vector_ptr_free_all` I get memory leak.

This should not happen if everything is working properly. Let's revert this to `vector_ptr_free_all` and figure out the true cause of the memory leak.

Reply:
o.k I found out what was my mistake... thanks!